apps/example/src/read-models/PayerReadModel.ts

```diff
@@ -1,38 +1,36 @@
-import { ITimelineEvent, StreamName } from "@equinox-js/core"
+import { StreamId } from "@equinox-js/core"
 import { Pool } from "pg"
 import { PayerId } from "../domain/identifiers.js"
 import { Payer } from "../domain/index.js"
-import { forEntity, Change, createProjection } from "@equinox-js/projection-pg"
+import { forEntity, Change, createHandler, MinimalClient } from "@equinox-js/projection-pg"
 
-type Payer = { id: PayerId; version: bigint; name: string; email: string }
+type Payer = { id: PayerId; name: string; email: string; version: string }
 
-const { Delete, Upsert } = forEntity<Payer, "id"| "version">()
+const { Delete, Upsert } = forEntity<Payer, "id" | "version">()
 
-export const projection = { table: "payer", id: ["id"], version: "version"}
+export const projection = { table: "payer", id: ["id"], version: "version" }
 
-function changes(stream: StreamName, events: ITimelineEvent[]): Change[] {
-  const id = Payer.Stream.tryMatch(stream) 
+type State = { name: string; email: string } | null
+
+function changes(streamId: StreamId, state: State, version: string): Change[] {
+  const id = Payer.Stream.decodeId(streamId)
   if (!id) return []
-  const event = Payer.Events.codec.decode(events[events.length - 1])
-  if (!event) return []
-  const version = events[events.length - 1]!.index
-  switch (event.type) {
-    case "PayerProfileUpdated":
-      const data = event.data
-      return [Upsert({ id: id, version, name: data.name, email: data.email })]
-    case "PayerDeleted":
-      return [Delete({ id: id, version })]
+  if (!state) return [Delete({ id, version })]
```

Comment:
hm actually deleting means replays would bring it back
but I guess making a tombstone to just to filter it out is a bit debatable too

Reply:
but remember, there are no replays, this happens as events are appended in the same transaction 🎉

Reply:
Yes, silly me!